jenkins/release-commit-lock/release-commit-lock.jenkinsfile

```diff
@@ -0,0 +1,168 @@
+                    def updateManifest = { String productName, String releaseCandidate ->
+                        def existingManifest = readYaml file: "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml"
+                        def selectedComponents = params.COMPONENTS.isEmpty() ? existingManifest.components : existingManifest.components.findAll { component ->
+                            params.COMPONENTS.split(',').any { it.trim() == component.name }
+                        }
+                        selectedComponents.each { componentName ->
+                            def existingComponent = existingManifest.components.find { it.name == componentName.name }
+                            if (existingComponent) {
+                                def releaseBranch = params.RELEASE_VERSION.split('\\.')[0..1].join('.')
+                                def repoHeadCommit = sh(script: "git ls-remote ${componentName.repository} ${releaseBranch} | cut -f 1", returnStdout: true).trim()
+                                if (repoHeadCommit) {
+                                    existingComponent.ref = repoHeadCommit
+                                }
+                            }
+                        }
+                        writeYaml file:  "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml", data: existingManifest, overwrite: true
+                        sh """
+                                yq eval -i '.' manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml
+                                sed -i '1s/^/---\\n/' manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml
+                        """
+                    }
+                    updateManifest("opensearch", params.OPENSEARCH_RELEASE_CANDIDATE)
+                    updateManifest("opensearch-dashboards", params.OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE)
+                }
+            }
+        }
+        stage('Create Pull Request') {
+            steps {
+                script {
+                    withCredentials([usernamePassword(credentialsId: 'jenkins-github-bot-token', passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
+                        try {
+                            sh """
+                                git remote set-url origin https://prudhvigodithi:${GITHUB_TOKEN}@github.com/prudhvigodithi/opensearch-build
+                                git config user.email "pgodithi@amazon.com"
+                                git config user.name "prudhvigodithi"
+                                git checkout -b manifest-lock
+                        """
+                            def status = sh(returnStdout: true, script: 'git status --porcelain')
+                            if (status) {
+                                sh """
+                                    git add . && git commit -sm "Manifest Commit Lock"
```

Comment:
Maybe grab the commit and PR title from params.Action? I believe it will be more clear whether it is locking loose commits or updating locked ones.

Reply:
PR titles lets have them generic Sayali (based on both the two actions), if passed as parameter it deviates from general titles and later if we have to gather these metrics would be hard.

Reply:
Updated with generic title that will show the type of action, based on `params.MANIFEST_LOCK_ACTION`.